src/frontend/Sider/Behandling/Inngangsvilkår/Målgruppe/EndreMålgruppeLæremidler.tsx

```diff
@@ -0,0 +1,233 @@
+
+    align-self: start;
+    align-items: start;
+`;
+
+const initaliserForm = (
+    eksisterendeMålgruppe?: Målgruppe,
+    registrertYtelsePeriode?: PeriodeYtelseRegister
+): EndreMålgruppeForm => {
+    return eksisterendeMålgruppe === undefined
+        ? nyMålgruppe(registrertYtelsePeriode)
+        : mapEksisterendeMålgruppe(eksisterendeMålgruppe);
+};
+
+export const EndreMålgruppeLæremidler: React.FC<{
+    målgruppe?: Målgruppe;
+    registerYtelsePeriode?: PeriodeYtelseRegister;
+    avbrytRedigering: () => void;
+}> = ({ målgruppe, avbrytRedigering, registerYtelsePeriode }) => {
+    const { behandling, behandlingFakta } = useBehandling();
+    const { oppdaterMålgruppe, leggTilMålgruppe, settStønadsperiodeFeil } = useInngangsvilkår();
+    const { lagreVilkårperiode } = useLagreVilkårperiode();
+
+    const [form, settForm] = useState<EndreMålgruppeForm>(
+        initaliserForm(målgruppe, registerYtelsePeriode)
+    );
+    const [laster, settLaster] = useState<boolean>(false);
+    const [feilmelding, settFeilmelding] = useState<Feil>();
+    const [vilkårsperiodeFeil, settVilkårsperiodeFeil] =
+        useState<FormErrors<MålgruppeValidering>>();
+
+    const delvilkårSomKreverBegrunnelse = finnBegrunnelseGrunnerMålgruppe(
+        form.type,
+        form.vurderinger
+    );
+    const begrunnelseErPåkrevd = delvilkårSomKreverBegrunnelse.length > 0;
+
+    const kanEndreType = målgruppe === undefined && registerYtelsePeriode === undefined;
+
+    const validerForm = (): boolean => {
+        const vilkårsperiodeFeil = validerMålgruppe(
+            form,
+            begrunnelseErPåkrevd,
+            målgruppe,
+            behandling.revurderFra
+        );
+        settVilkårsperiodeFeil(vilkårsperiodeFeil);
+
+        return isValid(vilkårsperiodeFeil);
+    };
+
+    const lagre = () => {
+        if (laster) return;
+        settFeilmelding(undefined);
+
+        const kanSendeInn = validerForm();
+
+        if (kanSendeInn) {
+            settLaster(true);
+
+            const erNyPeriode = målgruppe === undefined;
+
+            const response = lagreVilkårperiode<Målgruppe>(
+                behandling.id,
+                form,
+                mapFaktaOgSvarTilRequest(form),
+                målgruppe?.id
+            );
+            return response
+                .then((res) => {
+                    if (res.status === RessursStatus.SUKSESS) {
+                        if (erNyPeriode) {
+                            leggTilMålgruppe(res.data.periode);
+                        } else {
+                            oppdaterMålgruppe(res.data.periode);
+                        }
+                        if (res.data.stønadsperiodeStatus === StønadsperiodeStatus.Ok) {
+                            settStønadsperiodeFeil(undefined);
+                        } else if (res.data.stønadsperiodeFeil) {
+                            settStønadsperiodeFeil(lagFeilmelding(res.data.stønadsperiodeFeil));
+                        }
```

Comment:
Ikke helt koblet til PR, men fint å rydde opp i etterpå
Ser det finnes en util for dette som brukes for aktivitet. Den er ikke brukt for målgruppe fra før
```suggestion
                        oppdatertStønadsperiodeFeil(
                            res.data.stønadsperiodeStatus,
                            res.data.stønadsperiodeFeil
                        );
```